Split content of StringFunctions.h into binary, string, and url functions

isadikov opened the pull request:
This PR moves binary-related functions to `BinaryFunctions.h` and url-related functions to `URLFunctions.h` from `StringFunctions.h` according to the documentation, i.e. `binary.rst`, `string.rst`, and `url.rst`. Tests are also split in the same way - there is a total of original 56 tests split into 32 tests in StringFunctionsTest.cpp, 22 tests in BinaryFunctionsTest.cpp, and 2 tests in URLFunctionsTest.cpp.

This is a follow-up for https://github.com/facebookincubator/velox/pull/4281 where this separation was requested.

isadikov commented:
cc @aditi-pandit @mbasmanova I thought maybe we could just merge this PR, it is a simple split with all of the tests preserved (spread across several files). Thanks.

mbasmanova commented on velox/functions/prestosql/StringFunctions.h:
```diff
@@ -52,255 +52,6 @@ struct CodePointFunction {
 
-/// crc32(varbinary) → bigint
-/// Return an int64_t checksum calculated using the crc32 method in zlib.
-template <typename T>
```
Can we maybe remove some of the #includes now?

isadikov replied:
Updated.

mbasmanova commented on velox/functions/prestosql/URLFunctions.h:
```diff
@@ -18,6 +18,7 @@
 #include <boost/regex.hpp>
 #include <cctype>
 #include "velox/functions/Macros.h"
+#include "velox/functions/lib/string/StringImpl.h"
```
Let's remove this include and move urlEscape and urlUnescape to this file.

isadikov replied:
Updated.

mbasmanova commented on velox/functions/prestosql/tests/BinaryFunctionsTest.cpp:
```diff
@@ -0,0 +1,586 @@
+#include "velox/common/base/VeloxException.h"
+#include "velox/expression/Expr.h"
+#include "velox/functions/lib/StringEncodingUtils.h"
+#include "velox/functions/lib/string/StringImpl.h"
```
Do we need this include?

mbasmanova commented on velox/functions/prestosql/tests/BinaryFunctionsTest.cpp:
```diff
@@ -0,0 +1,586 @@
+#include "velox/functions/lib/string/StringImpl.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
```
Put the test into anonymous namespace inside facebook::velox::functions

isadikov replied:
Updated.

mbasmanova reviewed (commented):
@isadikov This is a nice improvement. Some comments.

isadikov commented:
@mbasmanova Could you take another look? I have addressed your comments and also inlined some of the functions from stringImpl into binary functions such as fromBase64, toBase64, md5, sha1, sha256, sha512, etc.

mbasmanova commented:
@isadikov Is this still a draft? If not, please, update.

isadikov commented:
Moved to Review mode, thanks.
